### ISSUE TITLE
Accept SVG import as a stable feature

### ISSUE THREAD

**Issue by kintel** [openscad/openscad#2702]
TODO:
* [ ] Decide on and document a minimal subset to support
* [x] Add more tests
* [ ] Call out any isolated bugs as separate issues
* [x] Test drag&drop of SVG files (tested on Linux)
* [ ] Test auto-reload and compile of imported SVG files

Supported SVG features:
* [x] Single shapes
* [x] Multiple shapes will be unioned
* [ ] FIXME: Document DPI/unit settings
* [ ] FIXME: Document how a viewport is imported
* [ ] FIXME: Document how stroke width influences imported shape

See also #1740, #1742, #2383, #2736

**Comment by MichaelPFrey** [comment]
I would like to ask, how you would like to handle errors/warnings within or related to libsvg.
(for e.g. file not found or parsing error)

Using printutils would go against the intention of it being standalone library. Leaves us with error codes or exceptions. Given that this is a c++ library and project, exceptions seam to be right mechanism.
Any thoughts? I have not yet looked deep into the code and I do not know the bigger picture, so I rather ask before starting to hack.

**Comment by t-paul** [comment]
#2596 adds a number of tests, mostly based on the SVG spec example files and also activates import/export test cases. I think that should cover the core functionality (= importing SVG Path).

As for error reporting, I suppose exceptions are the way to go. However, I don't think we need too much with regard to detailed warnings/parse errors messages as I suspect SVG is rarely written by hand (even though the spec seems to assume that is the main use case for some reason).

Note that #2596 also clarifies this bunch of code to have a different license (which was intended from the start but never happened to end up stated in the files). It probably makes sense have the relatively invasive changes from this PR merged first (or putting smaller changes on top of this branch).

**Comment by MichaelPFrey** [comment]
I do not want to add much error handling - but some like "file not found" and a generic "error during parsing" would make sense, as silently returning no geometry can cause hard to track down issues in bigger projects.

**Comment by t-paul** [comment]
Yes, that sounds great.

**Comment by t-paul** [comment]
closed via #2596

**Comment by t-paul** [comment]
Oh, whops, the experimental flag was not part of the PR...
